app/src/js/components/Conversations/conversation.js

```diff
@@ -20,66 +20,221 @@ import { AddAttachmentButton, DisplayAttachmentButton } from './attachment';
 
 
 const textRef = React.createRef();
-
-const getConversations = (dispatch, conversationId, lvl) => {
-  if (lvl) {
-    document.getElementById('all_button').classList.add('active');
-    document.getElementById('users_only_button').classList.remove('active');
-  } else {
-    document.getElementById('all_button').classList.remove('active');
-    document.getElementById('users_only_button').classList.add('active');
-  }
-  dispatch(getConversation(conversationId, lvl));
-};
-
-
-
-const Conversation = ({ dispatch, conversation, privileges, match }) => {
+const Conversation = ({ dispatch, conversation, privileges, match, user }) => {
   const current_user_id = JSON.parse(window.localStorage.getItem('auth-user')).id;
   const { conversationId } = match.params;
   const [showSearch, setShowSearch] = useState(false);
+  const [level, setLevel] = useState(false);
   const [uploadedFiles, setUploadedFiles] = useState([]);
-  useEffect(() => {
-    dispatch(getConversation(conversationId));
-  }, []);
+  // useEffect(() => {
+  //   dispatch(getConversation(conversationId));
+  // }, []);
+  const getConversations = async(dispatch, conversationId, lvl) => {
+    if (lvl) {
+      setLevel(lvl);
+      document.getElementById('all_button').classList.add('active');
+      document.getElementById('users_only_button').classList.remove('active');
+    } else {
+      setLevel(false);
+      document.getElementById('all_button').classList.remove('active');
+      document.getElementById('users_only_button').classList.add('active');
+    }
+
+    const newData = await dispatch(getConversation(conversationId, lvl));
+    if(newData?.data?.notes) setDisplayNotes(newData?.data?.notes);
+  };
   const { data, inflight, meta } = conversation;
-  const { subject, notes, participants } = data;
+  const { subject, notes = [], participants = [] } = data;
   const { queriedAt } = meta;
   const { canReply, canAddUser } = notePrivileges(privileges);
   const visibilityRef = useRef();
   const uploadedFilesRef = useRef();
 
+  const [tempNotes, setTempNotes] = useState([]);
+  const [displayNotes, setDisplayNotes] = useState(data.notes);
+  const [shouldStopRetries, setShouldStopRetries] = useState(false);
+
   const handleVisibilityReset = () => {
     visibilityRef?.current?.resetIdMap();
   };
 
+  const MAX_RETRIES = 7;
+  // Retry at 3s, 10s, 20s, 35s, 50s, 65s, 80s
+  const RETRY_INTERVALS = [3000, 5000, 10000, 15000, 15000, 15000, 15000];
```

Comment:
Comment says retry at 3s and 10s mark, but actually happening at 3s and 8s mark (3000+5000=8000) which offsets following retries slightly as well.